File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    private void dfs(int seq, int target, int[] numbers, List<Integer> result) {
        if (seq == numbers.length) {
            result.add(target == 0 ? 1 : 0);
            return;
        }
        
        dfs(seq + 1, target + numbers[seq], numbers, result);
        dfs(seq + 1, target - numbers[seq], numbers, result);
    }
    
    public int solution(int[] numbers, int target) {
        List<Integer> answer = new ArrayList<>();
        dfs(0, target, numbers, answer);
        return (int) answer.stream().filter(x -> x == 1).count();
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def dfs(seq, target, numbers, result):
    if seq == len(numbers):
        result.append(target == 0)
        return
    
    dfs(seq + 1, target + numbers[seq], numbers, result)
    dfs(seq + 1, target - numbers[seq], numbers, result)
    

def solution(numbers, target):
    answer = []
    dfs(0, target, numbers, answer)
    return sum(answer)
